crates/biome_js_analyze/src/lint/nursery/use_components_only_module.rs

```diff
@@ -0,0 +1,347 @@
+                    if let Some(exported_item_id) = &exported_item.identifier {
+                        if ctx
+                            .options()
+                            .allow_export_names
+                            .contains(&exported_item_id.text())
+                        {
+                            continue;
+                        }
+                    }
+                    // Allow exporting constants along with components
+                    if ctx.options().allow_constant_export
+                        && exported_item
+                            .exported
+                            .clone()
+                            .is_some_and(|partof| match partof {
+                                AnyJsExported::AnyJsExpression(expr) => {
+                                    expr.is_literal_expression()
+                                }
+                                _ => false,
+                            })
+                    {
+                        continue;
+                    }
+                    if is_exported_react_component(&exported_item) {
+                        exported_component_ids.push(exported_item);
+                    } else {
+                        exported_non_component_ids.push(exported_item);
+                    }
+                }
+            }
+        }
+
+        let local_component_ids = local_declaration_ids.iter().filter_map(|id| {
+            if Case::identify(&id.text(), false) == Case::Pascal {
+                Some(id.range())
+            } else {
+                None
+            }
+        });
+
+        if !exported_component_ids.is_empty() {
+            return exported_non_component_ids
+                .iter()
+                .filter_map(|id| {
+                    let range = id.identifier.clone().map(|x| x.range())?;
+                    Some(UseComponentsOnlyModuleState {
+                        error: ErrorType::ExportedNonComponentWithComponent,
+                        range,
+                    })
+                })
+                .collect::<Vec<UseComponentsOnlyModuleState>>();
+        }
+
+        local_component_ids
+            .map(|id| UseComponentsOnlyModuleState {
+                error: if exported_non_component_ids.is_empty() {
+                    ErrorType::UnexportedComponent
+                } else {
+                    ErrorType::NoExport
+                },
+                range: id,
+            })
+            .collect::<Vec<UseComponentsOnlyModuleState>>()
+    }
+
+    fn diagnostic(_ctx: &RuleContext<Self>, state: &Self::State) -> Option<RuleDiagnostic> {
+        match state.error {
+            ErrorType::ExportedNonComponentWithComponent => Some(
+                RuleDiagnostic::new(
+                    rule_category!(),
+                    state.range,
+                    markup! {
+                        "Export Non-Component with components are not allowed."
+                    },
+                )
```

Comment:
From the user's point of view, adding this consideration is more useful, I believe. 

```
               .note(markup! {
                    "Consider separating non-component exports into a new file." // Please update the description
                }),
```

Reply:
That was originally written in the `note` section, but I addressed it by splitting the long sentence for clarity.